lib/rubocop/cop/lint/redundant_cop_enable_directive.rb

```diff
@@ -82,14 +92,11 @@ def range_with_comma(comment, name)
           begin_pos = reposition(source, begin_pos, -1)
           end_pos = reposition(source, end_pos, 1)
 
-          comma_pos =
-            if source[begin_pos - 1] == ','
-              :before
-            elsif source[end_pos] == ','
-              :after
-            else
-              :none
-            end
+          comma_pos = find_comma_position(source, begin_pos, end_pos)
+
+          # If the list of cops is comma-separated, but without a white-space after the comma,
+          # we should **not** remove the prepending white-space, thus begin_pos += 1
+          begin_pos += 1 if comma_pos == :after && source[end_pos + 1] != ' '
```

Comment:
I think this logic should be moved into `range_to_remove`, no?

Reply:
@marcandre Thanks for reviewing. 
I think it doesn't matter much if the logic is here or in `range_to_remove`. We'd have to pass the variable `source` as an additional argument though, that's what made me put it here.
`source` is used to check if the character after the comma is a white space or not.

Reply:
The source is always available to cops, I'm not sure why it is being passed around. Actually I feel the cop is not particularly well written, using indices instead of `::Parser::Source::Range` (that also holds the source). If you are up for it, this cop would benefit from some refactoring, with `cop_name_indention => cop_name_range` for example. But at the least, please put that logic in `range_to_remove` as it is the responsibility of that method to calculate the range to be removed...

Reply:
@marcandre 👍 Fine for me, I've put the logic to `range_to_remove` as you indicated.
Also, I can give it a try to refactor the cop in the next days.

Reply:
@marcandre After moving the logic to `range_to_remove` it started to complain, of course, that the `Metrics/AbcSize` was too high. 🙈 Had to extract some logic once more.

Reply:
I'm wondering if combining `find_comma_position` and `range_to_remove` into one wouldn't make the whole thing simpler? Instead of `:after`, call a method `range_with_comma_after`, etc.? Otherwise put the code back in `range_to_remove` and add a `todo` for `AbcSize`; that metric will change anyways...